server/src/middleware/http/proxy_get_request.rs

```diff
@@ -53,29 +55,40 @@ pub struct InvalidPath(String);
 /// See [`ProxyGetRequest`] for more details.
 #[derive(Debug, Clone)]
 pub struct ProxyGetRequestLayer {
-	path: String,
-	method: String,
+	// path => method mapping
+	methods: Arc<FxHashMap<String, String>>,
```

Comment:
I think a plain `HashMap` would work as well, don't that would bring a noticeable performance hit 
```suggestion
	methods: Arc<HashMap<String, String>>,
```